test/xbgpu/test_engine.py

```diff
@@ -654,6 +805,28 @@ def heap_factory(batch_index: int) -> list[spead2.send.HeapReference]:
 
         assert actual_sensor_updates == expected_sensor_updates
 
+        # TODO (NGC-1169) have a separate command-line argument for this
+        channel_spacing = ADC_SAMPLE_RATE / n_samples_between_spectra
+        expected_beams = generate_expected_beams(
+            batch_start_index,
+            batch_end_index - batch_start_index,
+            n_channels_per_substream,
+            n_ants,
+            n_spectra_per_heap,
+            missing_antenna,
+            np.array([beam_output.pol for beam_output in beam_outputs]),
+            weights=weights,
+            delays=delays,
+            quant_gains=quant_gains,
+            channel_spacing=channel_spacing,
+            centre_channel=n_channels_total // 2 - frequency,
+        )
+        # assert_allclose converts to float, which bloats memory usage.
+        # To keep it manageable, compare a batch at a time.
```

Comment:
It does make the error messages less useful (because you only get to see the statistics on the particular slice you're comparing) so I don't recommend it in general. But I was running out of RAM on my machine without it.